Improve mypy config

alanhdu opened the pull request:
- Make mypy stricter
- Warn for unnecessary workarounds

alanhdu commented:
The `make check` failure should be fixed by https://github.com/ankitects/anki-desktop-i18n

dae commented on .github/scripts/contrib.sh:
```diff
@@ -9,7 +9,7 @@ git log --pretty=format:' - %ae' CONTRIBUTORS |sort |uniq |sort -f | sed "s/@/$a
 headAuthor=$(git log -1 --pretty=format:'%ae')
 authorAt=$(echo "$headAuthor" | sed "s/@/$antispam/")
-if git log --pretty=format:'%ae' CONTRIBUTORS | grep -qi "$headAuthor"; then
+if git log --pretty=format:'%ae' CONTRIBUTORS | grep -i "$headAuthor" > /dev/null; then
```
why did you change this?

alanhdu replied:
The CONTRIBUTOR check was failing otherwise:

https://github.com/ankitects/anki/runs/468917834

(I'll admit I don't really understand why -- I feel like it should work. Something about `pipefail` made the old version fail).

dae replied:
Ah, that makes sense thanks - grep will be closing the input as soon as it finds a line, which causes git log to lose its pipe.

dae commented on pylib/mypy.ini:
```diff
@@ -3,7 +3,12 @@ python_version = 3.7
 pretty = false
 no_strict_optional = true
 show_error_codes = true
+
```
Why the space?

alanhdu replied:
I thought it'd be easier to group the settings into "display settings" and "typecheck setings" (although I see now that I missed `no_strict_optional`). Is this something you'd want, or would you like me to revert the new line here?

dae replied:
python_version is not a display setting either - probably simplest to remove the gap.

dae commented:
The build is currently failing due to these changes. Are they working in your local environment? Maybe the hints have been improved in the latest PyQt release?

alanhdu commented:
@dae: I think I just messed up and didn't push the right things. I rebased on top of master to incorporate your change and removed the extra line, so this should be ready for another look!

dae commented:
Great, thanks!
